src/BackendStatusBanner/index.jsx

```diff
@@ -1,44 +1,48 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
 import statusMap from './statusMap.json';
+import { pingStudio } from '../data/actions/pingStudio';
 
 class BackendStatusBanner extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       apiConnectionStatus: 200,
     };
-
-    this.apiUrl = '/assets/course-v1:edX+DemoX+Demo_Course/?page=0&page_size=50&sort=sort&asset_type=';
-    if (!process.env.NODE_ENV || process.env.NODE_ENV === 'development') {
-      this.apiUrl = `/api${this.apiUrl}`;
-    }
   }
 
   componentDidMount() {
-    fetch(this.apiUrl, {
-      credentials: 'same-origin',
-      headers: {
-        Accept: 'application/json',
-      },
-    })
-      .then((response) => {
-        if (response.status !== this.state.apiConnectionStatus) {
-          this.setState({
-            apiConnectionStatus: response.status,
-          });
-        }
-      });
+    this.props.pingStudio();
   }
 
   render() {
-    return (this.apiConnectionStatus === 200) ?
+    return (this.props.connectionStatus === 200) ?
       null :
       (
         <div className="api-error">
-          {statusMap[this.state.apiConnectionStatus]}
+          {statusMap[this.props.connectionStatus]}
         </div>
       );
   }
 }
 
-export default BackendStatusBanner;
+BackendStatusBanner.propTypes = {
+  connectionStatus: PropTypes.number,
+  pingStudio: PropTypes.func.isRequired,
+};
+
+BackendStatusBanner.defaultProps = {
+  connectionStatus: null,
+};
+
+const WrappedBackendStatusBanner = connect(
+  state => ({
+    connectionStatus: state.connectionStatus,
+  }), dispatch => ({
+    pingStudio: () => dispatch(pingStudio()),
+  }),
+)(BackendStatusBanner);
+
+export default WrappedBackendStatusBanner;
```

Comment:
Good point! I wish we could, but you can't `export default const` ([this answer gives a pretty good explanation why](https://stackoverflow.com/a/36261387/1135244)). I'd be inclined to do something like
```
export default WrappedBackendStatusBanner = connect(
```
But because `WrappedBackendStatusBanner` contains a *class* and not a *function*, that syntax doesn't quite work. We'd have to declare an empty `let WrappedBackendStatusBanner;` further up the file, which I'm not personally a huge fan of.

Reply:
Errata: I'm wrong -- `WrappedBackendStatusBanner` does actually contain a function. It's the type of declaration (LexicalDeclaration) that makes things tricky here. That SO answer explains it much better than I can.